tst/EnergyPlus/unit/FaultFoulingAirFilters.unit.cc

```diff
@@ -0,0 +1,137 @@
+	bool TestRestult; 
+	
+	// Allocate
+	NumCurves = 1;
+	PerfCurve.allocate( NumCurves );
+	
+	NumFans = 2;
+	Fan.allocate( NumFans );
+		
+	// Inputs: fan curve
+	CurveNum = 1;
+	PerfCurve( CurveNum ).CurveType = Cubic;
+	PerfCurve( CurveNum ).ObjectType = CurveType_Cubic;
+	PerfCurve( CurveNum ).InterpolationType = EvaluateCurveToLimits;
+	PerfCurve( CurveNum ).Coeff1 = 1151.1;
+	PerfCurve( CurveNum ).Coeff2 = 13.509;
+	PerfCurve( CurveNum ).Coeff3 = -0.9105;
+	PerfCurve( CurveNum ).Coeff4 = -0.0129;
+	PerfCurve( CurveNum ).Coeff5 = 0.0;
+	PerfCurve( CurveNum ).Coeff6 = 0.0;
+	PerfCurve( CurveNum ).Var1Min = 7.0;
+	PerfCurve( CurveNum ).Var1Max = 21.0;
+	
+	// Inputs: fans
+	FanNum = 1;
+	Fan( FanNum ).FanName = "Fan_1";
+	Fan( FanNum ).FanType = "Fan:VariableVolume";
+	Fan( FanNum ).MaxAirFlowRate = 18.194;
+	Fan( FanNum ).DeltaPress = 1017.59;
+	
+	FanNum = 2;
+	Fan( FanNum ).FanName = "Fan_2";
+	Fan( FanNum ).FanType = "Fan:VariableVolume";
+	Fan( FanNum ).MaxAirFlowRate = 18.194;
+	Fan( FanNum ).DeltaPress = 1017.59 * 1.2;
+	
+	// Run and Check
+	// (1)The rated operational point of Fan_1 falls on the fan curve
+	TestRestult = CheckFaultyAirFilterFanCurve( "Fan_1", CurveNum );
+	EXPECT_TRUE( TestRestult ); 
+	// (2)The rated operational point of Fan_2 does not fall on the fan curve
+	TestRestult = CheckFaultyAirFilterFanCurve( "Fan_2", CurveNum );
+	EXPECT_FALSE( TestRestult ); 
+	
+	// Clean up
+	PerfCurve.deallocate( );
+	Fan.deallocate( ); 
+
+}
+
+TEST( FaultFoulingAirFilters, CalFaultyFanAirFlowReduction )
+{
+	// PURPOSE OF THIS SUBROUTINE:
+	// Calculate the decrease of the fan air flow rate, given the fan curve 
+	// and the increase of fan pressure rise due to fouling air filters
+		
+	int CurveNum;
+	int FanNum;
+	double FanDesignFlowRateDec;
+	double FanFaultyDeltaPressInc = 0.10; // Increase by 10%
+	
+	// Allocate
+	NumCurves = 1;
+	PerfCurve.allocate( NumCurves );
+	
+	NumFans = 1;
+	Fan.allocate( NumFans );
+		
+	// Inputs: fan curve
+	CurveNum = 1;
+	PerfCurve( CurveNum ).CurveType = Cubic;
+	PerfCurve( CurveNum ).ObjectType = CurveType_Cubic;
+	PerfCurve( CurveNum ).InterpolationType = EvaluateCurveToLimits;
+	PerfCurve( CurveNum ).Coeff1 = 1151.1;
+	PerfCurve( CurveNum ).Coeff2 = 13.509;
+	PerfCurve( CurveNum ).Coeff3 = -0.9105;
+	PerfCurve( CurveNum ).Coeff4 = -0.0129;
+	PerfCurve( CurveNum ).Coeff5 = 0.0;
+	PerfCurve( CurveNum ).Coeff6 = 0.0;
+	PerfCurve( CurveNum ).Var1Min = 7.0;
+	PerfCurve( CurveNum ).Var1Max = 21.0;
+	
+	// Inputs: fans
+	FanNum = 1;
+	Fan( FanNum ).FanName = "Fan_1";
+	Fan( FanNum ).FanType = "Fan:VariableVolume";
+	Fan( FanNum ).MaxAirFlowRate = 18.194;
+	Fan( FanNum ).DeltaPress = 1017.59;
+	
+	// Run and Check
+    FanDesignFlowRateDec = CalFaultyFanAirFlowReduction( Fan( FanNum ).FanName, Fan( FanNum ).MaxAirFlowRate, Fan( FanNum ).DeltaPress, 
+                           FanFaultyDeltaPressInc * Fan( FanNum ).DeltaPress, CurveNum );
+							
+	EXPECT_GE( FanDesignFlowRateDec, 3.84 ); 
```

Comment:
Sure, and you don't have to address all of these immediately.  I'd wait
until I get through the code so you don't have to make a ton of extra
commits.

On Wed, May 20, 2015 at 11:11 AM, Rongpeng Zhang <notifications@github.com>
wrote:

> In tst/EnergyPlus/unit/FaultFoulingAirFilters.unit.cc
> <https://github.com/NREL/EnergyPlus/pull/4894#discussion_r30724938>:
>
> > +	PerfCurve( CurveNum ).Coeff6 = 0.0;
> > +	PerfCurve( CurveNum ).Var1Min = 7.0;
> > +	PerfCurve( CurveNum ).Var1Max = 21.0;
> > +	
> > +	// Inputs: fans
> > +	FanNum = 1;
> > +	Fan( FanNum ).FanName = "Fan_1";
> > +	Fan( FanNum ).FanType = "Fan:VariableVolume";
> > +	Fan( FanNum ).MaxAirFlowRate = 18.194;
> > +	Fan( FanNum ).DeltaPress = 1017.59;
> > +	
> > +	// Run and Check
> > +    FanDesignFlowRateDec = CalFaultyFanAirFlowReduction( Fan( FanNum ).FanName, Fan( FanNum ).MaxAirFlowRate, Fan( FanNum ).DeltaPress,
> > +                           FanFaultyDeltaPressInc * Fan( FanNum ).DeltaPress, CurveNum );
> > +							
> > +	EXPECT_GE( FanDesignFlowRateDec, 3.84 );
>
> Thanks, @Myoldmopar <https://github.com/Myoldmopar> ! That's exactly what
> I needed but did not find.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NREL/EnergyPlus/pull/4894/files#r30724938>.
>



-- 
Edwin Lee, PhD
Engineer, Commercial Buildings Research Staff
National Renewable Energy Laboratory